src/sagemaker_training/mpi.py

```diff
@@ -95,18 +95,20 @@ def run(
 
         if wait:
             logger.info("Waiting for MPI process to finish.")
-            _wait_orted_process_to_finish()
+            self._wait_master_to_finish()
```

Comment:
I have a question (disclaimer: I have very limited knowledge on this topic, so it may be a dumb question)-- 
The check for connectivity to node-0 is via ssh now. if somehow ssh connectivity is broken temporarily (say sshd crashes and systemd will restart it), MPI will not be impacted since orted I assume will use a mechanism outside of ssh (say TCP sockets). in such a scenario can we falsely conclude that node-0 is done and exit worker processes?